lib/nolibc: Move prctl.h inside nolibc/musl-imported

TAGOliveira201000 opened the pull request:
Prctl.h is a libc header that needs to be inside libc so we avoid including it when using Musl.

<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [ ] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.


### Base target

 - Architecture(s): N/A
 - Platform(s): N/A
 - Application(s): any application using Musl 


### Additional configuration

<!--
Please specify any additional configuration which is needed for this feature to
work or any new configuration parameters which are introduced by this PR.  This
will help during the review process.  For example:

 - `CONFIG_LIBUKDEBUG=y`

-->

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

TAGOliveira201000 committed:
lib/nolibc: Move prctl.h inside nolibc/musl-imported

The "sys/prctl.h" header is specific to "nolibc". Move it from
"lib/posix-process/include/" to "lib/nolibc/musl-imported/include/".
This also fixes a warning caused by including it when Musl was used
(instead of using the Musl version).

Signed-off-by: Tiago Andre Goncalves Oliveira <t.g.oliveira@ua.pt>

razvand reviewed (changes requested):
Thanks. @TAGOliveira201000 . Please update the commit message to say something more descriptive like:

```
The `sys/prctl.h` header is specific to `nolibc`. Move it from `lib/posix-process/include/` to `lib/nolibc/musl-imported/include/`. This also fixes a warning caused by including it when Musl was used (instead of using the Musl version).
```

Also use your proper name and e-mail for signing off the commit.

Also, remove the `#ifdef UK_HAVE_SYSCALL` and `#endif` lines in the `sys/prctl.h` file. They are no longer required after moving it to nolibc.

TAGOliveira201000 committed:
lib/nolibc: Remove LIBC_SYSCALLS guard in prctl.h

sys/prctl.h in nolibc is using UK_LIBC_SYSCALLS as guard.
This is not required, since the guard only makes sense outside nolibc.
Remove UK_LIBC_SYSCALLS guard.

Signed-off-by: Tiago Andre Goncalves Oliveira <t.g.oliveira@ua.pt>

StefanJum reviewed (approved):
All good on my side.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

razvand reviewed (approved):
Approved-by: Razvan Deaconescu <razvand@unikraft.io>
